packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1205,18 +1218,7 @@ abstract class ResidentRunner {
   }
 
   @mustCallSuper
-  Future<void> preExit() async {
-    // If _dillOutputPath is null, we created a temporary directory for the dill.
-    if (_dillOutputPath == null && artifactDirectory.existsSync()) {
-      final File outputDill = globals.fs.file(dillOutputPath);
-      if (outputDill.existsSync()) {
-        outputDill.copySync(getDefaultCachedKernelPath(
-          trackWidgetCreation: trackWidgetCreation,
-        ));
-      }
-      artifactDirectory.deleteSync(recursive: true);
```

Comment:
Didn't this deletion pre-exist the cache.dill optimization? Why is it no longer needed?

Reply:
since it is a temp dir i gets cleaned up automatically right? Or should we still clean it out manually.

Reply:
I'd prefer still making an effort to do manual clean-up. The automatic cleanup doesn't work in all situations, for example if the tool is killed by a signal on Windows after `preExit()` runs.

Reply:
Makes sense, will add it back

Reply:
Done